File: bionic-js-tool/testing-code/swift/project-with-host-files/HostProject/host/libs/MotorVehicle.swift
import JavaScriptCore
import Bjs

class MotorVehicle: Vehicle {
    
    convenience init(_ weight: Int?, _ seats: Int?, _ maxSpeed: Int?, _ fuelType: FuelType?, _ maxRange: Double?, _ currentRange: Double?) {
        self.init(MotorVehicle.bjsClass, [MotorVehicle.bjs.putPrimitive(weight), MotorVehicle.bjs.putPrimitive(seats), MotorVehicle.bjs.putPrimitive(maxSpeed), MotorVehicle.bjs.putObj(fuelType), MotorVehicle.bjs.putPrimitive(maxRange), MotorVehicle.bjs.putPrimitive(currentRange)])
    }
    
    var isOnReserve:Bool? {
        get {
            return MotorVehicle.bjs.getBool(bjsGetProperty("isOnReserve"))
        }
    }
    
    var engine:Engine? {
        get {
            return MotorVehicle.bjs.getWrapped(bjsGetProperty("engine"), Engine.self)
        }
    }
    
    func refuel() -> Double? {
        return MotorVehicle.bjs.getFloat(bjsCall("refuel"))
    }
    
    func watchEngine(_ observer: (() -> String?)?) {
        let nativeFunc_bjs0 = observer
        let jsFunc_bjs1: @convention(block) () -> JSValue = {
            return MotorVehicle.bjs.putPrimitive(nativeFunc_bjs0!())
        }
        _ = bjsCall("watchEngine", MotorVehicle.bjs.putFunc(nativeFunc_bjs0, jsFunc_bjs1))
    }
    
    private static var _bjsLocator: BjsLocator = BjsLocator("BeautifulVehicles", "MotorVehicle")
    override class var bjsLocator: BjsLocator { _bjsLocator }
    override class func bjsFactory(_ jsObject: JSValue) -> MotorVehicle { MotorVehicle(jsObject) }
}

File: bionic-js-tool/testing-code/java/project-with-host-files/HostProject/src/motor-vehicles/java/test/project/host/libs/MotorVehicle.java
package test.project.host.libs;

import jjbridge.api.runtime.JSReference;
import jjbridge.api.value.strategy.FunctionCallback;
import bionic.js.Bjs;
import bionic.js.BjsAnyObject;
import bionic.js.BjsTypeInfo;
import bionic.js.BjsObjectTypeInfo;
import bionic.js.Lambda;
import java.util.Date;
import test.project.host.$native$.EngineBjsExport;
import test.project.host.libs.FuelType;
import test.project.host.libs.Vehicle;

@BjsTypeInfo.BjsLocation(project = "BeautifulVehicles", module = "MotorVehicle")
public class MotorVehicle extends Vehicle {
    
    protected <T extends Vehicle> MotorVehicle(Class<T> type, JSReference jsObject) {
        super(type, jsObject);
    }
    
    protected <T extends Vehicle> MotorVehicle(Class<T> type, JSReference[] arguments) {
        super(type, arguments);
    }
    
    public MotorVehicle(JSReference jsObject) {
        this(MotorVehicle.class, jsObject);
    }
    
    public MotorVehicle(JSReference[] arguments) {
        this(MotorVehicle.class, arguments);
    }
    
    public MotorVehicle(Long weight, Long seats, Long maxSpeed, FuelType fuelType, Double maxRange, Double currentRange) {
        this(bjs_MotorVehicle(weight, seats, maxSpeed, fuelType, maxRange, currentRange));
    }
    
    private static JSReference[] bjs_MotorVehicle(Long weight, Long seats, Long maxSpeed, FuelType fuelType, Double maxRange, Double currentRange) {
        return new JSReference[]{bjs.putPrimitive(weight), bjs.putPrimitive(seats), bjs.putPrimitive(maxSpeed), bjs.putObj(fuelType), bjs.putPrimitive(maxRange), bjs.putPrimitive(currentRange)};
    }
    
    public Boolean isOnReserve() {
        return bjs.getBoolean(bjsGetProperty("isOnReserve"));
    }
    
    public EngineBjsExport engine() {
        return bjs.getWrapped(bjsGetProperty("engine"));
    }
    
    public Double refuel() {
        return bjs.getDouble(bjsCall("refuel"));
    }
    
    public void watchEngine(Lambda.F0<String> observer) {
        Lambda.F0<String> nativeFunc_bjs0 = observer;
        FunctionCallback<?> jsFunc_bjs1 = jsReferences_bjs2 -> {
            jsReferences_bjs2 = bjs.ensureArraySize(jsReferences_bjs2, 0);
            return bjs.putPrimitive(nativeFunc_bjs0.apply());
        };
        bjsCall("watchEngine", bjs.putFunc(nativeFunc_bjs0, jsFunc_bjs1));
    }
    
    private static final JSReference bjsClass = BjsObjectTypeInfo.get(MotorVehicle.class).bjsClass();
    public static final Bjs bjs = BjsObjectTypeInfo.get(MotorVehicle.class).bjsLocator.get();
    public static final Bjs.JSReferenceConverter<MotorVehicle> bjsFactory = MotorVehicle::new;
}